pkg/controller/controller.go

```diff
@@ -471,10 +463,21 @@ func (c *Controller) runAction(ctx context.Context, t *tomb.Tomb, as *crv1alpha1
 			var msg string
 			if err == nil {
 				c.updateActionSetRunningPhase(ctx, as, p.Name())
+				progressTrackCtx, doneProgressTrack := context.WithCancel(ctx)
+				defer doneProgressTrack()
+				go func() {
+					// progress update is computed on a best-effort basis.
+					// if it exits with error, we will just log it.
+					if err := progress.UpdateActionSetsProgress(progressTrackCtx, c.crClient, as.GetName(), as.GetNamespace(), p); err != nil {
```

Comment:
Just wanted to understand that, we update the progress only when we start executing a phase? It's not continuously running in background and updating the status right?

Reply:
@viveksinghggits It is continuously running as a goroutine and checks for progress after every interval. And updates the status if there is any change in it

Reply:
@pavannd1 its shows nil progress if the phase is in pending state. Please check the samples I've added in the test plan.
E.g 
```
    phases:
    - name: phase1
      output:
        path: /operator-backups/2023-04-19T18-42-49/backup.tgz
      progress:
        lastTransitionTime: "2023-04-19T18:43:10Z"
        progressPercent: "100"
      state: complete
    - name: phase2
      progress: {}
      state: running
    - name: phase3
      progress: {}
      state: pending
  error:
    message: ""
  progress:
    lastTransitionTime: "2023-04-19T18:43:10Z"
    percentCompleted: "33"
    runningPhase: phase2
  state: running
```